server/etcdserver/api/etcdhttp/health.go

```diff
@@ -250,29 +290,33 @@ func (reg *CheckRegistry) runHealthChecks(ctx context.Context, checkNames ...str
 		}
 		if err := check(ctx); err != nil {
 			fmt.Fprintf(&individualCheckOutput, "[-]%s failed: %v\n", checkName, err)
-			h.Health = "false"
+			h.Status = HealthStatusError
+			recordMetrics(reg.checkType, checkName, HealthStatusError)
 		} else {
 			fmt.Fprintf(&individualCheckOutput, "[+]%s ok\n", checkName)
+			recordMetrics(reg.checkType, checkName, HealthStatusSuccess)
 		}
 	}
 	h.Reason = individualCheckOutput.String()
 	return h
 }
 
 // installRootHttpEndpoint installs the http handler for the root path.
-func (reg *CheckRegistry) installRootHttpEndpoint(lg *zap.Logger, mux *http.ServeMux, path string, checks ...string) {
-	hfunc := func(r *http.Request) Health {
+func (reg *CheckRegistry) installRootHttpEndpoint(lg *zap.Logger, mux *http.ServeMux, checks ...string) {
+	hfunc := func(r *http.Request) HealthStatus {
 		// extracts the health check names to be excludeList from the query param
 		excluded := getQuerySet(r, "exclude")
 
 		filteredCheckNames := filterCheckList(lg, listToStringSet(checks), excluded)
-		return reg.runHealthChecks(r.Context(), filteredCheckNames...)
+		h := reg.runHealthChecks(r.Context(), filteredCheckNames...)
+		recordMetrics(reg.checkType, "root", h.Status)
```

Comment:
Why special "root" check name? It's just a sum of other checks, I also didn't find anything like this in Kubernetes code.

Reply:
It is true the root metrics do not exist in Kubernetes code. 
Imaging I just want to set up an alert based on the overall health status, in theory that could be computed based on the min of all of the other checks, but I could easily make a mistake of using `max` instead of `min`. I think it could be convenient to have a root metric.

Reply:
But, if we are following Kubernetes design which expects users to utilize `min` function, why should we expect users to write different queries. Would it be more convenient for users to utilize same query across all components?